libraries/AP_InertialSensor/FastRateBuffer.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <AP_AHRS/AP_AHRS.h>
+#include "AP_InertialSensor_rate_config.h"
+#include "AP_InertialSensor.h"
+#include "AP_InertialSensor_Backend.h"
+
+#if AP_INERTIALSENSOR_FAST_SAMPLE_WINDOW_ENABLED
+#include "FastRateBuffer.h"
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+// hal.console can be accessed from bus threads on ChibiOS
+#define debug(fmt, args ...)  do {hal.console->printf("IMU: " fmt "\n", ## args); } while(0)
+#else
+#define debug(fmt, args ...)  do {printf("IMU: " fmt "\n", ## args); } while(0)
+#endif
+
+// enable the fast rate buffer and start pushing samples to it
+void AP_InertialSensor::enable_fast_rate_buffer()
+{
+    if (fast_rate_buffer_enabled) {
+        return;
+    }
+    if (fast_rate_buffer == nullptr) {
+        fast_rate_buffer = NEW_NOTHROW FastRateBuffer();
+    }
+    fast_rate_buffer_enabled = true;
```

Comment:
Should we check the buffer was allocated before enabling ?
```suggestion
    fast_rate_buffer_enabled = fast_rate_buffer != nullptr;
```

Reply:
Fixed